test/TokenTestUtils.js

```diff
@@ -229,23 +252,24 @@ async function checkVariables(_tokens, _customVars) {
         var token = _tokens[n];
         var customVars = _customVars[n];
 
-        let expectedState = buildExpectedState(token, customVars);
+        let expectedState = mapBNToHex(buildExpectedState(token, customVars));
+        let actualState = mapBNToHex(await getActualState(token));
         if (debugLogging) {
            console.log(util.inspect(expectedState, { showHidden: false, depth: null }))
+           console.log(util.inspect(actualState, { showHidden: false, depth: null }))
         }
 
-        let actualState = await getActualState(token);
         assertDiff.deepEqual(actualState, expectedState, "difference between expected and actual state");
 
         // Check that sum of individual balances equals totalSupply
         var accounts = Object.keys(Accounts).map(accountName => Accounts[accountName]);
         var balanceSum = bigZero;
         var x;
         for (x = 0; x < accounts.length; x++) {
-            balanceSum = balanceSum.plus(new BigNumber(await token.balanceOf(accounts[x])));
+            balanceSum = balanceSum.add(await token.balanceOf(accounts[x]));
         }
-        var totalSupply = new BigNumber(await token.totalSupply())
-        assert(balanceSum.isEqualTo(totalSupply), "sum of balances is not equal to totalSupply");
+        var totalSupply = newBigNumber(await token.totalSupply())
+        assert(balanceSum.cmp(totalSupply)==0, "sum of balances is not equal to totalSupply");
```

Comment:
Minor, but maybe we want to check for equals here instead of comparison, so we don't have to check == 0 ?

Reply:
The new big number library BN does not have an .equals function, so I had to change to `.cmp(x)==0 everywhere`.  I agree it is very ugly.

Reply:
Will change this everywhere

Reply:
The BN library does not work with assert.equals - it returns false even if the numbers are equal because the internal representation might be different.  This is one of the reasons I had to to use the hex representation for the deepEquals function.